### ISSUE TITLE
Compatibility profiles

### ISSUE THREAD

**Issue by stianst** [keycloak/keycloak#35650]
### Description

We currently have the ability to define different versions of a feature when backwards compatibility is broken; but we don't have a server-wide approach for when it's not wrapped in a feature.

Additionally, anyone that wants to stay on the latest and greatest would need to bump all feature versions.

We should consider introducing something like `--feature=compatible|latest|preview`, where:

* Compatible - is guaranteed to not introduce any breaking changes
* Latest - introduces breaking changes and uses the latest supported versions of a feature
* Preview - enables all preview features, switches to preview feature versions, etc.

Additionally, some behaviour not wrapped in a feature could use the compatibility profile to change behaviour. We may also want to remove/disable some deprecated capabities in `latest`.

Basically the aim here is that people can choose whether or not they want to:

* Stay compatible and not making any changes until upgrading to 27
* Adopt changes as they come, getting ready early for 27
* Preview new stuff (not recommended for production ;))

**Comment by shawkins** [comment]
From a greenfield perspective would you want to stay in latest as long as possible?

When would you drop from latest to compatible? If it were a feature related issue would you instead selectively override that feature's version? 

If instead you hit a compatibility issue that was unrelated to any feature and want to drop from latest to compatible does that imply the only unrelated changes allowed via latest would be removals of deprecated options - because any other changes would potentially be breaking for someone going from latest to compatible.

**Comment by thomasdarimont** [comment]
As far as I can tell, these new profiles become effective once users roll out new Keycloak versions. What about giving users the information earlier, e.g., showing compatibility with the next Keycloak version upfront in the admin UI?

A Keycloak server could periodically check for new Keycloak versions online (which would need to be explicitly enabled by the server admin). The keycloak website could provide an updated metadata endpoint that contains information about the feature compatibility across versions. With this, Keycloak admins can get notified about new versions with additional compatibility information.
